apiserver/Volumes.md

```diff
@@ -0,0 +1,147 @@
+# Volumes support for Ray cluster by the API server
+
+API server allows to specify multiple types of volumes mounted to the Ray pods (nodes). These include:
+[hostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath), 
+[PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/),
+[ephemeral](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/),
+[config maps](https://kubernetes.io/docs/concepts/storage/volumes/#configmap),
+[secrets](https://kubernetes.io/docs/concepts/storage/volumes/#secret),
+and [empty dir](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir).
+Multiple volumes of different type can be mounted to both head and worker nodes, by defining a volume array for them 
+
+
+## HostPath volumes
+
+A hostPath volume mounts a file or directory from the host node's filesystem into your Pod. This is not something that 
+most Pods will need, but it offers a powerful escape hatch for some applications.
+
+For example, some uses for a hostPath are:
+
+* running a container that needs access to Docker internals; use a hostPath of /var/lib/docker
+* running cAdvisor in a container; use a hostPath of /sys
+* allowing a Pod to specify whether a given hostPath should exist prior to the Pod running, whether it should be 
+created, and what it should exist as
+
+The code below gives an example of hostPath volume definition:
+
+````
+{
+	"name": "hostPath",             # unique name
+	"source": "/tmp",               # data location on host
+	"mountPath": "/tmp/hostPath",   # mounting path
+	"volumeType": 1,                # volume type - host path
+	"hostPathType": 0,              # host path type - directory
+	"mountPropagationMode": 1       # mount propagation - host to container
+}
+````
+
+## PVC volumes
+
+A Persistent Volume Claim (PVC) is a request for storage by a user. It is similar to a Pod. Pods consume node resources 
+and PVCs consume PV resources. Pods can request specific levels of resources (CPU and Memory). Claims can request 
+specific size and access modes (e.g., they can be mounted `ReadWriteOnce`, `ReadOnlyMany` or `ReadWriteMany`).
+
+The caveat of using PVC volumes is that the same PVC is mounted to all nodes. As a result only PVCs with access
+mode `ReadOnlyMany` can be used in this case.
+
+The code below gives an example of PVC volume definition:
+
+````
+{
+	"name": "pvc",              # unique name
+	"mountPath": "/tmp/pvc",    # mounting path
+	"volumeType": 0,            # volume type - PVC
+	"mountPropagationMode": 2,  # mount propagation mode - bidirectional
+	"readOnly": false           # read only
```

Comment:
Since we talked about access mode, maybe we could also show an example
```suggestion
	"readOnly": false           # read only,
	"accessMode": "ROX" # readOnlyMany
```

Reply:
Its not specified here, sorry